bin/commonmarker

```diff
@@ -45,16 +57,38 @@ ARGV.delete_if do |arg|
       exit 1
     end
     true
+  elsif arg =~ /^--list-parse-options$/
+    puts parse_options.keys
+    exit 0
+  elsif arg =~ /^--parse-option=(.+)$/
+    if parse_options.key?($1.to_sym)
```

Comment:
Heya, so there's two things I'd want to change here:

1. The ability to pass in a comma separated list of keys
2. A test to validate this arg passing behavior

I can get to this if you don't want to, but it'll take me about a week as I'm a bit busy on some other things right now. Just putting it out there!

Reply:
I agree, I considered a comma separated list, just figured I would keep the interface you have.  I'm happy to make the change.  But wondering, do mind if I rewrite it using [OptionParser](https://ruby-doc.org/stdlib-2.5.1/libdoc/optparse/rdoc/OptionParser.html) ?

I don't mind adding test.  Just to make sure I'm not missing something obvious, I don't see any tests at the moment testing the command line interface?

Reply:
Yes, unfortunately no tests exist... I don’t mind writing the test myself actually. And yeah adding a proper arg parsing dependency would be swell. Thanks!

Reply:
I've converted to using `OptionParser`, and the nice thing is it gives the comma separated options basically for free.  Take a look and let me know what you think.

Not sure I have time to get to the tests right now, would be happy if you have time for that.  Otherwise I would try to swing around to it in a few days.